### DISCUSSION TITLE
Could not poll the database: SQLSTATE[HY000] [2002]

### DISCUSSION THREAD

**Discussion by lokesh-krishna** [firefly-iii/firefly-iii#7064] [Firefly III] [answered]
Attempting to get Firefly running on my Ubuntu 22.04.1 VPS. Followed the instructions for the docker-compose route and after running `docker-compose -f docker-compose.yml up`, I get the following error when navigating to Firefly on my browser:

```

Whoops! An error occurred.

Unfortunately, this error was not recoverable :(. Firefly III broke. The error is:

Could not poll the database: SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'172.19.0.3' (using password: YES) (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)
```

My docker-compose.yml: [docker-compose.yml](https://paste.sr.ht/~low-key/71f2d47a99287462b13eb714981714213712f5c0)
My .env: [.env](https://paste.sr.ht/~low-key/04053ab26903b9f1caa030df91f4432cbb5b7021)

Would appreciate any help in figuring this out. Is there any additional information I can provide to help troubleshoot this?

**Comment by JC5** [comment, marked as answer]
People ask this every other day, please check out similar discussions for your answer :)

**Reply by lokesh-krishna** [reply]
Thanks for replying. I did check before posting and tried several of the solutions without them helping. Could I ask for just a little of your time to take a look and see if I'm missing anything obvious?

**Reply by JC5** [reply]
Sure, but I'm not sure yet what you tried. 

Usually, this is happens when you change the database password after the database has started for the first time. So make sure you delete all volumes and then start the compose file again to make sure the old database (and old password) is removed.

**Reply by JC5** [reply]
https://docs.firefly-iii.org/firefly-iii/faq/docker/#the-database-password-is-wrong-but-im-100-sure-its-correct

**Reply by lokesh-krishna** [reply]
Thanks a lot! I did try removing both, the container and the volume but still kept facing the same issue. Turns out I had a typo in my `.env` file, with `DB_HOST` set to `fireflyiidb` instead of `fireflyiiidb`. Sorry about the noise and thanks for all of your patience.

**Reply by JC5** [reply]
Ohhh, that's a tricky one, nice find!
